cli/lsp/language_server.rs

```diff
@@ -3031,6 +3031,36 @@ impl Inner {
 
 #[tower_lsp::async_trait]
 impl tower_lsp::LanguageServer for LanguageServer {
+  async fn execute_command(
+    &self,
+    params: ExecuteCommandParams,
+  ) -> LspResult<Option<Value>> {
+    if params.command == "deno.cache" {
+      let mut arguments = params.arguments.into_iter();
+      let uris: Vec<Url> = arguments
+        .next()
+        .and_then(|value| serde_json::from_value(value).ok())
+        .ok_or(LspError::invalid_params("missing or incorrect uris"))?;
+      let referrer: Url = arguments
+        .next()
+        .and_then(|value| serde_json::from_value(value).ok())
+        .ok_or(LspError::invalid_params("missing or incorrect referrer"))?;
```

Comment:
This lets us maximally defer error messages to serde:
```suggestion
      let uris = serde_json::to_value(arguments.next()).unwrap();
      let uris: Vec<Url> = serde_json::from_value(uris)
        .map_err(|err| LspError::invalid_params(err.to_string()))?;
      let referrer = serde_json::to_value(arguments.next()).unwrap();
      let referrer: Url = serde_json::from_value(referrer)
        .map_err(|err| LspError::invalid_params(err.to_string()))?;
```